PIMbench/cpp-relu/PIM/relu.cpp

```diff
@@ -259,72 +254,74 @@ int main(int argc, char *argv[]) {
   int inputWidth = inputMatrix[0][0].size();
 
   // Define parameters for processing
-  unsigned numCols = 8192, numOfCore = 4096;
-  int numOfPIMRow = 1;
-  int numOfPIMColumn = (inputHeight * inputWidth / numOfPIMRow);
-  int numOfMatPerRow = std::min(static_cast<int>(std::floor((1.0 * numCols * numOfCore) / numOfPIMColumn)), inputDepth);
+  PimDeviceProperties deviceProp;
+  PimStatus status = pimGetDeviceProperties(&deviceProp);
+  if (status != PIM_OK) {
+    std::cout << "Abort: pimGetDeviceProperties failed" << std::endl;
+    return 1;
+  }
+  // Get the device parameters
+  uint64_t numCols = deviceProp.numColPerSubarray;
+  uint64_t numRows = deviceProp.numRowPerSubarray;
+  uint64_t numOfBits = uint64_t(deviceProp.numRanks) * uint64_t(deviceProp.numBankPerRank) * uint64_t(deviceProp.numSubarrayPerBank) * numCols * numRows;
+
+  uint64_t numOfPIMRow = 1;
+  uint64_t numOfMatPerRow = std::min(static_cast<uint64_t>(std::floor((1.0 * numOfBits) / (inputHeight * inputWidth * 32))), static_cast<uint64_t>(inputWidth));
```

Comment:
Is `32` here the size of data type?

Reply:
Yes, 32 is the size of the data type. Instead of hardcoding it, declared it as a variable, and updated the file accordingly.